### PR TITLE
Add Pagination to Android

### PR DESCRIPTION
**Pull request by leeminkyu16**
Not a perfect implementation.
Will follow up with optimizations.

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt [hunk 1]
```diff
@@ -104,14 +104,33 @@ fun HomePageView(
...
+				if (listState.layoutInfo.totalItemsCount == 0) {
+					false
+				}
+				else {
+					listState.layoutInfo.visibleItemsInfo.lastOrNull()?.index == listState.layoutInfo.totalItemsCount - 1
+				}
+			}
+		}
+
+		LaunchedEffect(key1 = lastItemIsShowing) {
```
Comment (andr-dev) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt:121]: Just out of curiosity, what is key1 here?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt [hunk 2]
```diff
@@ -140,4 +184,8 @@ class HomePageViewModel @Inject constructor(
 			it.dispose()
 		}
 	}
+
+	companion object {
+		const val LISTING_PAGE_SIZE = 5
```
Comment (andr-dev) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt:189]: We can use 5 for now, should probably bump this up to 10 in the future.

**Review by andr-dev — APPROVED** [review]
Nice!

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt:121]: `key1` basically means that the side effect in `LaunchedEffect` will run every time the `key1` is updated.

Not the optimal solution probably, but it works for now.

**Review by abhaykshar — APPROVED** [review]
huge !

**Pull request merged**